### PR TITLE
Code cleanups

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

File: okio/jvm/jmh/src/jmh/java/com/squareup/okio/benchmarks/IndexOfElementBenchmark.java [hunk 1]
```diff
@@ -40,7 +40,7 @@
 @BenchmarkMode(Mode.SampleTime)
 @OutputTimeUnit(TimeUnit.MICROSECONDS)
 public class IndexOfElementBenchmark {
-  ByteString byteString = ByteString.encodeUtf8("abcd");
+  final ByteString byteString = ByteString.encodeUtf8("abcd");
```
Comment (swankjesse) [inline okio/jvm/jmh/src/jmh/java/com/squareup/okio/benchmarks/IndexOfElementBenchmark.java:43]: I don’t think `final` on locals in Java is a good tradeoff for legibility vs. safety

File: okio/src/commonMain/kotlin/okio/-Base64.kt [hunk 2]
```diff
@@ -87,8 +87,7 @@ internal fun String.decodeBase64ToArray(): ByteArray? {
     }
   }
 
-  val lastWordChars = inCount % 4
-  when (lastWordChars) {
+  when (inCount % 4) {
```
Comment (swankjesse) [inline okio/src/commonMain/kotlin/okio/-Base64.kt:90]: This loses a useful name, lastWordChars

File: okio/src/commonMain/kotlin/okio/internal/-Buffer.kt [hunk 3]
```diff
@@ -1297,7 +1296,7 @@ internal inline fun Buffer.commonIndexOf(bytes: ByteString, fromIndex: Long): Lo
     while (offset < resultLimit) {
       // Scan through the current segment.
       val data = s.data
-      val segmentLimit = okio.minOf(s.limit, s.pos + resultLimit - offset).toInt()
```
Comment (swankjesse) [inline okio/src/commonMain/kotlin/okio/internal/-Buffer.kt:1300]: This one was interesting; we had our own minOf for mixed parameter types. Is this no longer necessary?

File: okio/src/commonMain/kotlin/okio/internal/-SegmentedByteString.kt [hunk 4]
```diff
@@ -115,10 +115,9 @@ internal inline fun SegmentedByteString.commonSubstring(beginIndex: Int, endInde
 
   val newSegments = segments.copyOfRange(beginSegment, endSegment + 1)
   val newDirectory = IntArray(newSegments.size * 2)
-  var index = 0
-  for (s in beginSegment..endSegment) {
+  for ((index, s) in (beginSegment..endSegment).withIndex()) {
     newDirectory[index] = minOf(directory[s] - beginIndex, subLen)
-    newDirectory[index++ + newSegments.size] = directory[s + segments.size]
+    newDirectory[index + newSegments.size] = directory[s + segments.size]
```
Comment (swankjesse) [inline okio/src/commonMain/kotlin/okio/internal/-SegmentedByteString.kt:120]: Good find here

File: okio/src/hashFunctions/kotlin/okio/internal/Md5.kt [hunk 5]
```diff
@@ -95,8 +95,7 @@ internal class Md5 : HashFunction {
     var d = h3
 
     for (i in 0 until 16) {
-      val g = i
-      val f = ((b and c) or (b.inv() and d)) + a + localK[i] + words[g]
+      val f = ((b and c) or (b.inv() and d)) + a + localK[i] + words[i]
```
Comment (swankjesse) [inline okio/src/hashFunctions/kotlin/okio/internal/Md5.kt:98]: I think this is closer to the spec when defined longform

File: okio/src/jvmTest/java/okio/ByteStringJavaTest.java [hunk 6]
```diff
@@ -92,7 +93,7 @@
         new Object[] { Factory.ONE_BYTE_PER_SEGMENT, "SegmentedByteString (one-at-a-time)" });
   }
 
-  @Parameter(0) public Factory factory;
+  @Parameter() public Factory factory;
```
Comment (swankjesse) [inline okio/src/jvmTest/java/okio/ByteStringJavaTest.java:96]: I think this is more clear with the literal 0

**Review by swankjesse — CHANGES_REQUESTED** [review]
Thanks, but I don’t think this is a net improvement. Most of the time when there’s style warnings we’re deliberately disobeying them because we disagree with the tools emitting those warnings.

File: okio/src/commonMain/kotlin/okio/internal/-Buffer.kt (hunk 3 above)
Comment (Goooler) [reply okio/src/commonMain/kotlin/okio/internal/-Buffer.kt:1300]: Just simplify the package name, it has been imported.

File: okio/jvm/jmh/src/jmh/java/com/squareup/okio/benchmarks/IndexOfElementBenchmark.java (hunk 1 above)
Comment (Goooler) [reply okio/jvm/jmh/src/jmh/java/com/squareup/okio/benchmarks/IndexOfElementBenchmark.java]: Reverted.

File: okio/src/commonMain/kotlin/okio/-Base64.kt (hunk 2 above)
Comment (Goooler) [reply okio/src/commonMain/kotlin/okio/-Base64.kt]: Reverted.

File: okio/src/jvmTest/java/okio/ByteStringJavaTest.java (hunk 6 above)
Comment (Goooler) [reply okio/src/jvmTest/java/okio/ByteStringJavaTest.java]: Reverted.

**Pull request closed without merging**